libflux/src/semantic/analyze.rs

```diff
@@ -0,0 +1,2580 @@
+use crate::ast;
+use crate::semantic::nodes::*;
+use crate::semantic::types::{Fresher, MonoType};
+use std::result;
+
+type SemanticError = String;
+type Result<T> = result::Result<T, SemanticError>;
+
+// analyze analyzes an AST package node and returns its semantic analysis.
+// The function requires a Fresher to instantiate fresh type veraibles for Expressions.
+// The function explicitely moves the ast::Package because it adds information to it.
+// We follow here the principle that every compilation step should be isolated and should add meaning
+// to the previous one. In other terms, once one analyzes an AST he should not use it anymore.
+// If one wants to do so, he should explicitely pkg.clone() and incur consciously in the memory
+// overhead involved.
+pub fn analyze(pkg: ast::Package, fresher: &mut Fresher) -> Result<Package> {
+    analyze_package(pkg, fresher)
+    // TODO(affo): run checks on the semantic graph.
+}
+
+fn type_info(fresher: &mut Fresher) -> MonoType {
+    MonoType::Var(fresher.fresh())
+}
```

Comment:
I think we can replace calls to `type_info` now with `MonoType::Var(fresher.fresh())` to be more explicit.